platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -134,6 +134,14 @@ typedef void (^MGLOfflinePackAdditionCompletionHandler)(MGLOfflinePack * _Nullab
  */
 typedef void (^MGLOfflinePackRemovalCompletionHandler)(NSError * _Nullable error);
 
+/**
+ A block to be called once the contents of a file are copied into the current packs.
+ 
+ @param error Contains a pointer to an error object (if any) indicating why the
+ pack could not be created or added.
+ */
+typedef void (^MGLOfflinePacksAdditionCompletionHandler)(NSError * _Nullable error);
```

Comment:
> Likewise, completion handlers should always include context about the operation that has completed.

For this case specifically I don't consider it would be useful (although I agree with you about the context) pass the packs that were added/updated, because those are not the same packs after we reload the packs. For the test case the contents of the `barcelona.db` returns 4 packages which you can't explicitly know which were updated/added. Since there is no relation between the packs that are stored and the ones that this api returns as updated/added I considered this to mislead the developer to think it would be possible to perform any action on the returned packs.

Reply:
What about the `mbgl::OfflineRegion`s that `DefaultFileSource::mergeOfflineRegions()` passes into its callback? Correct me if I’m wrong, but these are the regions in the merged database that originate from the sideloaded database:

https://github.com/mapbox/mapbox-gl-native/blob/6e171a62a9cc07ddbb0765e3021e46ebd60313df/platform/default/mbgl/storage/offline_database.cpp#L691-L695

For the purposes of distinguishing between multiple uses of a callback, I think it would additionally be useful for this completion handler to accept the file URL or file path of the sideloaded database.

/cc @asheemmamoowala

Reply:
> these are the regions in the merged database that originate from the sideloaded database:

@1ec5 - these regions may be newly added or updated regions. The completion handler can return the set of packs in the SDK binding, but I don't think that is useful enough by itself. For Observers of the `MGLOfflineStorage.packs` property will not be able to tell which packs were added vs updated. If the internal array is simply reloaded, again they observers would lose context of what mutations have occurred to the array.

Making KVO compliant changes to the `packs` property seems like a valuable change. I think that would look like: 
1. adding any new packs to the `packs` property, similarly to how `addPack` does it: `[[strongSelf mutableArrayValueForKey:@"packs"] addObject:pack];`
 2. Matching region ids returned from core to existing packs and invalidating those OR replacing them with new objects using `[NSMutableArray replaceObjectAtIndex]` or some other KVO compliant method.